digitalocean/reservedipv6/resource_reserved_ipv6_assignment_test.go

```diff
@@ -0,0 +1,197 @@
+package reservedipv6_test
+
+import (
+	"context"
+	"fmt"
+	"regexp"
+	"strconv"
+	"testing"
+
+	"github.com/digitalocean/godo"
+	"github.com/digitalocean/terraform-provider-digitalocean/digitalocean/acceptance"
+	"github.com/digitalocean/terraform-provider-digitalocean/digitalocean/config"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/resource"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/terraform"
+)
+
+func TestAccDigitalOceanReservedIPV6Assignment(t *testing.T) {
+	var reservedIPv6 godo.ReservedIPV6
+
+	resource.ParallelTest(t, resource.TestCase{
+		PreCheck:          func() { acceptance.TestAccPreCheck(t) },
+		ProviderFactories: acceptance.TestAccProviderFactories,
+		CheckDestroy:      testAccCheckDigitalOceanReservedIPV6Destroy,
+		Steps: []resource.TestStep{
+			{
+				Config: testAccCheckDigitalOceanReservedIPV6AssignmentConfig,
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheckDigitalOceanReservedIPV6AttachmentExists("digitalocean_reserved_ipv6_assignment.foobar"),
+					resource.TestMatchResourceAttr(
+						"digitalocean_reserved_ipv6_assignment.foobar", "ip", regexp.MustCompile(ipv6Regex)),
+					resource.TestMatchResourceAttr(
+						"digitalocean_reserved_ipv6_assignment.foobar", "droplet_id", regexp.MustCompile("[0-9]+")),
+				),
+			},
+
```

Comment:
Could you please include a tests for reassigning. 
Something like this
```suggestion
			{
				Config: testAccCheckDigitalOceanReservedIPV6AssignmentReassign,
				Check: resource.ComposeTestCheckFunc(
					testAccCheckDigitalOceanReservedIPV6AttachmentExists("digitalocean_reserved_ipv6_assignment.foobar"),
					resource.TestMatchResourceAttr(
						"digitalocean_reserved_ipv6_assignment.foobar", "ip", regexp.MustCompile(ipv6Regex)),
					resource.TestMatchResourceAttr(
						"digitalocean_reserved_ipv6_assignment.foobar", "droplet_id", regexp.MustCompile("[0-9]+")),
				),
			},
```
Some variables can be updated for the test. This may look like:

```

var testAccCheckDigitalOceanReservedIPV6AssignmentConfig = `
resource "digitalocean_reserved_ipv6" "foobar" {
  region_slug = "nyc3"
}

resource "digitalocean_droplet" "foobar1" {
  name   = "foobar1"
  size   = "s-1vcpu-1gb"
  image  = "ubuntu-22-04-x64"
  region = "nyc3"
  ipv6   = true
}

resource "digitalocean_reserved_ipv6_assignment" "foobar" {
  ip         = digitalocean_reserved_ipv6.foobar.ip
  droplet_id = digitalocean_droplet.foobar1.id
}
`
var testAccCheckDigitalOceanReservedIPV6AssignmentReassign = `
resource "digitalocean_reserved_ipv6" "foobar" {
  region_slug = "nyc3"
}

resource "digitalocean_droplet" "foobar1" {
  name   = "foobar1"
  size   = "s-1vcpu-1gb"
  image  = "ubuntu-22-04-x64"
  region = "nyc3"
  ipv6   = true
}

resource "digitalocean_reserved_ipv6_assignment" "foobar" {
  ip         = digitalocean_reserved_ipv6.foobar.ip
  droplet_id = digitalocean_droplet.foobar2.id
}

resource "digitalocean_droplet" "foobar2" {
  name               = "foobar2"
  size               = "s-1vcpu-1gb"
  image              = "ubuntu-22-04-x64"
  region             = "nyc3"
  ipv6               = true
}
`

var testAccCheckDigitalOceanReservedIPV6AssignmentDeleteAssignment = `
resource "digitalocean_reserved_ipv6" "foobar" {
  region_slug = "nyc3"
}

resource "digitalocean_droplet" "foobar1" {
  name   = "foobar1"
  size   = "s-1vcpu-1gb"
  image  = "ubuntu-22-04-x64"
  region = "nyc3"
  ipv6   = true
}

resource "digitalocean_droplet" "foobar2" {
  name               = "foobar2"
  size               = "s-1vcpu-1gb"
  image              = "ubuntu-22-04-x64"
  region             = "nyc3"
  ipv6               = true
}
`
```

Reply:
@loosla at API level we don't support Reassign

Reply:
Could you add a test that is showing how the users should UnAssign and then Assign to a different droplet :pray: 
Thank you